mmv1/third_party/terraform/tests/resource_container_node_pool_test.go.erb

```diff
@@ -3003,3 +3003,84 @@ resource "google_container_node_pool" "np2" {
   }
 `, cluster, np1, np2)
 }
+
+func testAccContainerNodePool_withSoleTenantConfig(cluster, np string) string {
+       return fmt.Sprintf(`
+data "google_container_engine_versions" "central1a" {
+  location = "us-central1-a"
+}
+
+resource "google_compute_node_template" "soletenant-tmpl" {
+  name      = "tf-test-soletenant-tmpl"
+  region    = "us-central1"
+  node_type = "n1-node-96-624"
+}
+
+resource "google_compute_node_group" "nodes" {
+  name        = "tf-test-soletenant-group"
+  zone        = "us-central1-a"
+  size          = 1
+  node_template = google_compute_node_template.soletenant-tmpl.id
+}
+
+resource "google_container_cluster" "cluster" {
+  name               = "%s"
+  location           = "us-central1-a"
+  initial_node_count = 1
+  min_master_version = data.google_container_engine_versions.central1a.latest_master_version
+}
+
+resource "google_container_node_pool" "with_sole_tenant_config" {
+  name               = "%s"
+  location           = "us-central1-a"
+  cluster            = google_container_cluster.cluster.name
+  initial_node_count = 1
+  node_config {
+       machine_type = "n1-standard-2"
+       sole_tenant_config {
+			node_affinity {
+				key = "compute.googleapis.com/node-group-name"
+				operator = "IN"
+				values = [google_compute_node_group.nodes.name]
+			}
+       }
+    oauth_scopes = [
+      "https://www.googleapis.com/auth/logging.write",
+      "https://www.googleapis.com/auth/monitoring",
+    ]
+  }
+}
+`, cluster, np)
+}
+
+func TestAccContainerNodePool_withSoleTenantConfig(t *testing.T) {
```

Comment:
nit: Do you mind grouping this with the other acceptance test cases further up in the file? Older files are structured with acceptance test functions first, and configs later.